@here/harp-mapview/test/PoiRendererTest.ts

```diff
@@ -97,4 +103,107 @@ describe("PoiRenderer", function () {
             expect(screenBox.h).to.equal(16);
         });
     });
+
+    describe("search for cached images", function () {
+        const webGLRenderer = {
+            capabilities: {
+                isWebGL2: false
+            }
+        } as THREE.WebGLRenderer;
+        const mapViewStub = sinon.createStubInstance(MapView);
+        const poiManager = new PoiManager((mapViewStub as any) as MapView);
+        const testImageName = "testImage";
+        const mapEnv = new Env();
+
+        const createPointLabel = (name: string) => {
+            return {
+                poiInfo: {
+                    imageTextureName: name,
+                    isValid: true,
+                    buffer: undefined,
+                    technique: {}
+                },
+                visible: true
+            } as TextElement;
+        };
+
+        let x = 1;
+        let y = 1;
+
+        const addRandomImageToCache = (
+            testCache: MapViewImageCache,
+            name: string,
+            load: boolean
+        ) => {
+            // Note, the images must be unique, otherwise the test will fail, because the internal
+            // caching mechanism of the ImageCache will have already loaded the image.
+            return testCache.addImage(name, `https://picsum.photos/${x++}/${y++}`, load);
+        };
+
+        it("poi is valid when in cache and loading started", async function () {
+            this.timeout(5000);
+
+            const testCache = new MapViewImageCache();
+            const testImageItem = addRandomImageToCache(testCache, testImageName, true);
+            const caches = [testCache];
+            const poiRenderer = new PoiRenderer(webGLRenderer, poiManager, caches);
+            const pointLabel = createPointLabel(testImageName);
+            const waitingForLoad = poiRenderer.prepareRender(pointLabel, mapEnv);
+            // This is false because the image is still being loaded in the background, notice the
+            // true flag passed to the call to testCache.addImage.
+            expect(waitingForLoad).false;
+
+            // Promise.resolve used to convert the type `ImageItem | Promise<ImageItem>` to
+            // `Promise<ImageItem>`.
+            await Promise.resolve(testImageItem);
+
+            const imageLoaded = poiRenderer.prepareRender(pointLabel, mapEnv);
+            expect(imageLoaded).true;
```

Comment:
Why does this pass? Looking at the code, it semes that preparePoi would return false at line 675, cause poinInfo.isValid would have been set to false in the previous call.